### PR TITLE
chore: Hiero renaming initial

### PR DESCRIPTION
**Pull request by gsstoykov**
**Description**:
This PR introduces Hiero renaming for the GO SDK.

**Related issue(s)**:

Fixes https://github.com/hiero-ledger/hiero-sdk-go/issues/1081

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: .github/CODEOWNERS
Comment (rbarkerSL) [inline .github/CODEOWNERS:1]: My suggested updates

```text
# Default code owners for entire repository
*                                               @hiero-ledger/hiero-sdk-go-maintainers @hiero-ledger/hiero-sdk-go-committers
#########################
#####  Core Files  ######
#########################
# NOTE: Must be placed last to ensure enforcement over all other rules

# Protection Rules for Github Configuration Files and Actions Workflows
/.github/                                       @hiero-ledger/github-maintainers @hiero-ledger/github-committers
/.github/workflows/                             @hiero-ledger/github-maintainers @hiero-ledger/github-committers @hiero-ledger/hiero-sdk-go-maintainers @hiero-ledger/hiero-sdk-go-committers

# Legacy Maven project files
**/pom.xml                                      @hiero-ledger/github-maintainers @hiero-ledger/hiero-sdk-go-maintainers @hiero-ledger/hiero-sdk-go-committers

# Gradle project files and inline plugins
/gradle/                                        @hiero-ledger/github-maintainers @hiero-ledger/github-committers @hiero-ledger/hiero-sdk-go-maintainers @hiero-ledger/hiero-sdk-go-committers
gradlew                                         @hiero-ledger/github-maintainers @hiero-ledger/github-committers @hiero-ledger/hiero-sdk-go-maintainers @hiero-ledger/hiero-sdk-go-committers
gradlew.bat                                     @hiero-ledger/github-maintainers @hiero-ledger/github-committers @hiero-ledger/hiero-sdk-go-maintainers @hiero-ledger/hiero-sdk-go-committers
**/build-logic/                                 @hiero-ledger/github-maintainers @hiero-ledger/github-committers @hiero-ledger/hiero-sdk-go-maintainers @hiero-ledger/hiero-sdk-go-committers
**/gradle.*                                     @hiero-ledger/github-maintainers @hiero-ledger/github-committers @hiero-ledger/hiero-sdk-go-maintainers @hiero-ledger/hiero-sdk-go-committers
**/*.gradle.*                                   @hiero-ledger/github-maintainers @hiero-ledger/github-committers @hiero-ledger/hiero-sdk-go-maintainers @hiero-ledger/hiero-sdk-go-committers

# Codacy Tool Configurations
/config/                                        @hiero-ledger/github-maintainers @hiero-ledger/hiero-sdk-go-maintainers @hiero-ledger/hiero-sdk-go-committers
.remarkrc                                       @hiero-ledger/github-maintainers @hiero-ledger/hiero-sdk-go-maintainers @hiero-ledger/hiero-sdk-go-committers

# Self-protection for root CODEOWNERS files (this file should not exist and should definitely require approval)
/CODEOWNERS                                     @hiero-ledger/github-maintainers

# Protect the repository root files
/README.md                                      @hiero-ledger/github-maintainers @hiero-ledger/hiero-sdk-go-maintainers @hiero-ledger/hiero-sdk-go-committers @hiero-ledger/tsc
**/LICENSE                                      @hiero-ledger/github-maintainers @hiero-ledger/tsc
```

**Review by rbarkerSL — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml
Comment (rbarkerSL) [inline .github/workflows/main.yml:1]: Nit: whitespace got shifted dramatically.

The hashgraph standard for yaml files is two space characters ' ' per indent. I also see that the character `'` has been replaced by `"` throughout the file. Recommend reverting that.

File: .github/CODEOWNERS
Comment (rbarkerSL) [inline .github/CODEOWNERS:18]:
```suggestion
gradlew                                         @hiero-ledger/github-maintainers @hiero-ledger/github-committers
gradlew.bat                                     @hiero-ledger/github-maintainers @hiero-ledger/github-committers
```

**Review by rbarkerSL — CHANGES_REQUESTED** [review]
Recommend addressing merge conflicts on main.yml and updating to the latest. I believe

**Review by 0xivanov — APPROVED** [review]
Reviewed Go files.

**Review by rbarkerSL — APPROVED** [review]
Approval applies to the following files:

- .github/CODEOWNERS
- .github/workflows/main.yml
- CHANGELOG.md
- LICENSE
- README.md
- go.mod
- tck/go.mod

I would like to here from @hendrikebbers or @nathanklick on if the change to the license file should have removed the hashgraph copyright or not? but this LGTM

**Comment by gsstoykov** [comment]
@rbarkerSL In C++ SDK we got it in with removing the Hashgraph copyright. We had some discussions around that topic. Will merge the PR if something comes up I will open a followup to change it.

**Pull request merged**
